internal/provider/datasources/work_queue.go

```diff
@@ -0,0 +1,172 @@
+		return
+	}
+
+	client, ok := req.ProviderData.(api.PrefectClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected api.PrefectClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+}
+
+var workQueueAttributes = map[string]schema.Attribute{
+	"id": schema.StringAttribute{
+		Computed:    true,
+		CustomType:  customtypes.UUIDType{},
+		Description: "Work queue ID (UUID)",
+		Optional:    true,
+	},
+	"created": schema.StringAttribute{
+		Computed:    true,
+		CustomType:  customtypes.TimestampType{},
+		Description: "Date and time of the work queue creation in RFC 3339 format",
+	},
+	"updated": schema.StringAttribute{
+		Computed:    true,
+		CustomType:  customtypes.TimestampType{},
+		Description: "Date and time that the work queue was last updated in RFC 3339 format",
+	},
+	"work_pool_name": schema.StringAttribute{
+		Computed:    true,
+		CustomType:  customtypes.TimestampType{},
+		Description: "The name of the work pool that the work queue belongs to",
+	},
+	"name": schema.StringAttribute{
+		Computed:    true,
+		Description: "Name of the work queue",
+		Optional:    true,
+	},
+	"description": schema.StringAttribute{
+		Computed:    true,
+		Description: "Description of the work queue",
+		Optional:    true,
+	},
+	"is_paused": schema.BoolAttribute{
+		Computed:    true,
+		Description: "Whether this work queue is paused",
+	},
+	"concurrency_limit": schema.Int64Attribute{
+		Computed:    true,
+		Description: "The concurrency limit applied to this work queue",
+		Optional:    true,
+	},
+	"priority": schema.Int64Attribute{
+		Computed:    true,
+		Description: "The work queue's priority",
+		Optional:    true,
+	},
+}
+
+// Schema defines the schema for the data source.
+func (d *WorkQueueDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: `
+Get information about an existing Work Queue by name.
+<br>
+Use this data source to obtain Work Queue-specific attributes.
+`,
+		Attributes: workQueueAttributes,
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (d *WorkQueueDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var model WorkQueueDataSourceModel
+
+	// Populate the model from data source configuration and emit diagnostics on error
+	resp.Diagnostics.Append(req.Config.Get(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client, err := d.client.WorkQueues(model.AccountID.ValueUUID(), model.WorkspaceID.ValueUUID(), model.WorkPoolName.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating variable client",
+			fmt.Sprintf("Could not create variable client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
```

Comment:
```suggestion
			"Error creating work queue client",
			fmt.Sprintf("Could not create work queue client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
```

Reply:
ok I was curious about this - we reference `variable client` in a few other go files. I will update them on this branch